### PR TITLE
OCPBUGS-1520: Fixes SNAT-ing Logic for EgressIPs

### PR DESCRIPTION
**Pull request by tssurya**
Conflicts for each commit are outlined in commit description.
/cc @anuragthehatter let's start the QE verifications
/cc @jechen0648 

/cc @trozet
/cc @kyrtapz 
/cc @flavio-fernandes 

I also brought in the e2e tests that Jaime added for egressIP gRPC healthcheck feature, shouldn't matter since we don't run e2e's downstream anyways and it made my conflict life easier.

### CODE REVIEW

**Comment by tssurya** [comment]
/hold
want https://github.com/openshift/origin/blob/master/test/extended/networking/egressip.go these to run successfully

**Comment by flavio-fernandes** [comment]
@tssurya : remove hold?

**Comment by tssurya** [comment]
@flavio-fernandes @trozet PTAL

File: go-controller/pkg/ovn/obj_retry.go [hunk 1]
```diff
@@ -1278,14 +1278,14 @@ func (oc *Controller) getSyncResourcesFunc(r *RetryObjs) (func([]interface{}) er
 	case factory.EgressFirewallType:
 		syncFunc = oc.syncEgressFirewall
 
-	case factory.EgressIPType:
```
Comment (trozet) [inline go-controller/pkg/ovn/obj_retry.go:1281]: @ricky-rav remember when you backport the retry refactor to make sure this is right

**Review by trozet — APPROVED** [review]
/lgtm

**Comment by trozet** [comment]
ci/prow/e2e-aws-ovn-local-to-shared-gateway-mode-migration - failed because northd inactivity timer fired for 39 minutes - 
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ovn-kubernetes/1331/pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-local-to-shared-gateway-mode-migration/1584919173480321024

**Comment by trozet** [comment]
/override ci/prow/e2e-aws-ovn-local-to-shared-gateway-mode-migration

**Pull request merged**